src/api_helpers.jl

```diff
@@ -254,6 +254,12 @@ function h5s_get_simple_extent_dims(space_id)
     h5s_get_simple_extent_dims(space_id, dims, maxdims)
     return dims, maxdims
 end
+function h5s_get_simple_extent_dims(space_id, ::Nothing)
```

Comment:
I don't think there's a good way to avoid this right?

Doing something like:
```julia
function h5s_get_simple_extent_dims(space_id; maxdims = false)
           n = h5s_get_simple_extent_ndims(space_id)
           dims = Vector{hsize_t}(undef, n)
           if maxdims
               mdims = Vector{hsize_t}(undef, n)
               h5s_get_simple_extent_dims(space_id, dims, mdims)
               return dims, mdims
           else
               h5s_get_simple_extent_dims(space_id, dims, C_NULL)
               return dims
           end
end
```
would make the return type unstable.

Reply:
Yeah, that's why I just went with a separate method. I went with `nothing` argument since `C_NULL = Ptr{Nothing}(0)`, so _(vigorous hand waving)_ they're kind of similar-ish.